src/frontend/src/lib/services/address.services.ts

```diff
@@ -179,23 +187,43 @@ export const certifyAddress = async (address: string): Promise<ResultSuccess<str
 			return { success: false, err: 'Using the dapp with an anonymous user if not supported.' };
 		}
 
-		const certifiedAddress = await getEthAddress(identity);
+		const certifiedAddress = await getAddress(identity);
 
 		if (address.toLowerCase() !== certifiedAddress.toLowerCase()) {
 			return {
 				success: false,
-				err: 'The address used to load the data did not match your actual wallet address, which is why your session was ended. Please sign in again to reload your own data.'
+				err: `The address used to load the data did not match your actual ${tokenId.description} wallet address, which is why your session was ended. Please sign in again to reload your own data.`
 			};
 		}
 
 		addressStore.set({ tokenId, data: { data: address, certified: true } });
 
-		await updateIdbEthAddressLastUsage(identity.getPrincipal());
+		await updateIdbAddressLastUsage(identity.getPrincipal());
 	} catch (err: unknown) {
 		addressStore.reset(tokenId);
 
-		return { success: false, err: 'Error while loading the ETH address.' };
+		return { success: false, err: `Error while loading the ${tokenId.description} address.` };
 	}
 
 	return { success: true };
 };
+
+export const certifyBtcAddressMainnet = async (address: string): Promise<ResultSuccess<string>> =>
+	certifyAddress({
+		tokenId: BTC_MAINNET_TOKEN_ID,
+		address,
+		getAddress: (identity) =>
```

Comment:
`identity` has no type here? Typescript inherits it?

Reply:
indeed it inherits it, that is why it didn't raise an error. But I will put it explicits just to be sure